crates/snforge-scarb-plugin/src/types.rs

```diff
@@ -51,9 +51,24 @@ impl ParseFromExpr<Expr> for Felt {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
 pub struct Number(pub(crate) BigInt);
 
+impl Number {
+    pub fn validate_in_gas_range<T: AttributeInfo>(
+        &self,
+        arg_name: &str,
+    ) -> Result<(), Diagnostic> {
+        let max = u64::MAX;
+        if *self > Number(max.into()) {
+            Err(T::error(format!(
+                "{arg_name} it too large (max permissible value is {max})"
+            )))?;
+        }
```

Comment:
Shouldn't we also verify that the user doesn't provide a negative value? Or is that impossible?

Reply:
it's impossible - it's on scarb plugin level afaik (if you try that, some error diagnostic will be raised as far as I remember)